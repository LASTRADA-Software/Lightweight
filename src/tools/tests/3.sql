CREATE TABLE IF NOT EXISTS User (
    id              INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    fullname        VARCHAR(128) NOT NULL,
    email           VARCHAR(60) NOT NULL
);
CREATE TABLE IF NOT EXISTS TaskList (
    id              INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    user_id         INT NOT NULL,
    CONSTRAINT      fk1 FOREIGN KEY (user_id) REFERENCES user(id)
);
CREATE TABLE IF NOT EXISTS TaskListEntry (
    id              INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    tasklist_id     INT NOT NULL,
    completed       DATETIME NULL,
    task            VARCHAR(255) NOT NULL,
    CONSTRAINT      fk1 FOREIGN KEY (tasklist_id) REFERENCES TaskList(id)
);
