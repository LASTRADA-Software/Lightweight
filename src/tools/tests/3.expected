// SPDX-License-Identifier: Apache-2.0
#pragma once
#include <Lightweight/DataMapper/DataMapper.hpp>
#include <Lightweight/SqlConnection.hpp>
#include <Lightweight/SqlDataBinder.hpp>
#include <Lightweight/SqlQuery.hpp>
#include <Lightweight/SqlQueryFormatter.hpp>
#include <Lightweight/SqlScopedTraceLogger.hpp>
#include <Lightweight/SqlStatement.hpp>
#include <Lightweight/SqlTransaction.hpp>

struct TaskList;
struct TaskListEntry;
struct User;

struct User final
{
    Field<int, PrimaryKey::AutoIncrement> id;
    Field<std::string> fullname;
    Field<std::string> email;
};

struct TaskList final
{
    Field<int, PrimaryKey::AutoIncrement> id;
    BelongsTo<&User::id> user;
};

struct TaskListEntry final
{
    Field<int, PrimaryKey::AutoIncrement> id;
    Field<std::optional<SqlDateTime>> completed;
    Field<std::string> task;
    BelongsTo<&TaskList::id> taskList;
};
